### PR TITLE
Friends - MultiSession : Replace close() by quit() in WebDriver and quit all sessions after test

### PR DESCRIPTION
**Pull request by eXorus**
Hello,

I'm using Selenium Grid with Codeception Friends and Codeception doesn't kill correctly the browser so I'm stuck with a lot of node used for nothing :

```
public function testFriend1(AcceptanceTester $I)
{
   $I->amOnUrl('http://localhost/toto-nofriend');

   $toto = $I->haveFriend('toto', $I);
   $toto->does(function($I) {
      $I->amOnUrl('http://localhost/toto-friend');
   });
}
```

with Selenium Hub (2.53) and 2 Firefox nodes
"http://localhost/toto-nofriend" will be on node1
"http://localhost/toto-friend" will be on node2

**At the end of the test with close()**
node1 will be free slot
node2 will be always used (with state deleted but still used)

**At the end of the test with quit()**
node1 will be free slot
node2 will be free slot

#3039 #2836

### CODE REVIEW

**Comment by eXorus** [comment]
Fix 11 issues from scrutinizer. I'm done with this PR.

**Comment by eXorus** [comment]
I found an issue when test failed (element not visible for example). The PR is not ready yet

**Comment by eXorus** [comment]
I'm done with this PR, it's working well for me on local selenium and selenium grid @DavertMik @Naktibalda 

Actually Codeception kill the browsers of the friends at the end of the PHP script (when all objects are destructs).
This PR saved all the sessions opened during a run (friends or not) and kill them at the end of the test case or suite according to the configuration.

**Comment by DavertMik** [comment]
Thank you very much!
This is awesome.

File: src/Codeception/Module/WebDriver.php [hunk 1]
```diff
@@ -394,10 +392,21 @@ protected function formatLogEntries(array $logEntries)
     public function _afterSuite()
     {
         // this is just to make sure webDriver is cleared after suite
-        if (isset($this->webDriver)) {
-            $this->webDriver->quit();
+        $this->cleanWebDriver();
+    }
+
+    public function cleanWebDriver()
```
Comment (DavertMik) [inline src/Codeception/Module/WebDriver.php]: I will make it protected

**Pull request merged**